packages/core/src/components/NetworkSwitch/NetworkSwitch.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import { CHAIN_ID_TO_NETWORK } from '@web3-ui/hooks';
+import { Select, SelectProps } from '@chakra-ui/react';
+
+const useChainId = () => {
+  const [chainId, setChainId] = useState<null | string>(null);
+  useEffect(() => {
+    window.ethereum.on('chainChanged', setChainId);
+    window.ethereum.request({ method: 'eth_chainId' }).then(setChainId);
+  }, []);
+  return chainId;
+};
```

Comment:
Curious whether folks think we should just add this to the hooks package

Reply:
That's a great idea

Reply:
This only works with Metamask tho, right?

Reply:
@Dhaiwat10 yeah, just MetaMask

Reply:
Just reading your twitter post about the project's future. Should we then keep this `useChainId()` hook integrated with the `<NetworkSwitch/>` component, because we'll be pivoting away from hooks in general?

Reply:
I went ahead and added it for now